### ISSUE TITLE
E3002 Invalid Property Resources/FilesBucket/Properties/ReplicationConfiguration/Rules/0/ID

### ISSUE THREAD

**Issue by ville-nvt** [aws-cloudformation/cfn-lint#2454]
### CloudFormation Lint Version

0.69.1

### What operating system are you using?

ubuntu

### Describe the bug

Regarding S3/ReplicationConfiguration/ReplicationRule AWS documentation states:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-replicationconfiguration-rules.html
`Id
...**When using a V2 replication configuration this property is capitalized as "ID"**.`

Using ID instead of Id is considered error

### Expected behavior

ID should be allowed parameter

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  ExampleBucket:
    Type: AWS::S3::Bucket
    Properties:
      BucketName: "example-bucket-name"
      ReplicationConfiguration:
        Role: !GetAtt ReplicationRole.Arn
        Rules:
          - ID: "ExampleBackupRule"
            Filter:
              Prefix: ""
            Status: Enabled
            Destination:
              Bucket: !GetAtt  DestinationBucket.Arn
              Account: 123456789
              StorageClass: "STANDARD_IA"
              AccessControlTranslation:
                Owner: "Destination"
              ReplicationTime:
                Status: "Enabled"
                Time:
                  Minutes: 15
              Metrics:
                Status: "Enabled"
                EventThreshold:
                  Minutes: 15
            DeleteMarkerReplication:
              Status: "Disabled"

  DestinationBucket:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: "destination-bucket-name"
                
  ReplicationRole:
    Type: "AWS::IAM::Role"
    Properties:
      RoleName: "ExampleReplicationRole"
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
          - Effect: "Allow"
            Principal:
              Service: "s3.amazonaws.com"
            Action:
              - "sts:AssumeRole"
      Policies:
        - PolicyName: "ExampleReplicationReadPolicy"
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: "Allow"
                Action:
                  - "s3:Get*"
                  - "s3:ListBucket"
                Resource:
                  - "arn:aws:s3:::example-bucket-name"
                  - "arn:aws:s3:::example-bucket-name/*"
        - PolicyName: "ExampleReplicationWritePolicy"
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - Effect: "Allow"
                Action:
                  - "s3:ReplicateObject"
                  - "s3:ReplicateDelete"
                  - "s3:ReplicateTags"
                Resource: "arn:aws:s3:::destination-bucket-name/*"
```
